drivers/sensor/lis2mdl/lis2mdl_spi.c

```diff
@@ -122,6 +122,13 @@ int lis2mdl_spi_init(struct device *dev)
 		cfg->gpio_cs_port, cfg->cs_gpio);
 #endif
 
+#if CONFIG_LIS2MDL_SPI_FULL_DUPLEX
+	/* Set SPI 4wires */
+	if (lis2mdl_spi_mode_set(data->ctx, LIS2MDL_SPI_4_WIRE) < 0) {
```

Comment:
why do this twice (once in `lis2mdl_init` and again here)?

Reply:
@MaureenHelm 
Are you ok with this explanation?